common/Message.hpp

```diff
@@ -161,14 +159,21 @@ class Message
         return (forward.find('-') != std::string::npos ? forward : std::string());
     }
 
-    const char* skipWhitespace(const char* p)
+    std::vector<char> copyDataAfterOffset(const char *p, size_t len, size_t fromOffset)
     {
-        while (p && *p == ' ')
+        if (!p || fromOffset >= len)
+            return std::vector<char>();
+
+        size_t i;
+        for (i = fromOffset; i < len; ++i)
         {
-            ++p;
+            if (p[i] != ' ')
+                break;
         }
-
-        return p;
+        if (i < len)
+            return std::vector<char>(p + i, p + len - i);
```

Comment:
The second argument is the last item pointer, not the length. So should be `p + len`.